torch/_inductor/kernel/mm.py

```diff
@@ -741,6 +963,116 @@ def tuned_sparse_semi_structured_mm(
     )
 
 
+@register_lowering(aten._scaled_mm.default, type_promotion_kind=None)  # type: ignore[misc]
+def tuned_scaled_mm(
+    mat_a,
+    mat_b,
+    scale_a,
+    scale_b,
+    bias=None,
+    scale_result=None,
+    out_dtype=None,
+    use_fast_accum=False,
+    layout=None,
+):
+    from torch._inductor.select_algorithm import realize_inputs
+
+    m, n, k, layout, mat_a, mat_b = mm_args(
+        mat_a, mat_b, layout=layout, out_dtype=out_dtype
+    )
+    # below is for getting an overview logging info of inductor mms
+    counters["aten_mm_info"][f"aten._scaled_mm.default_{m}_{n}_{k}"] += 1
+    log.info(
+        "Tuned aten._scaled_mm.default: m=%s, n=%s, k=%s, mat1_dtype=%s, mat2_dtype=%s, output_layout=%s",
+        m,
+        n,
+        k,
+        mat_a.get_dtype(),
+        mat_b.get_dtype(),
+        layout,
+    )
+
+    check_supported_striding(mat_a, mat_b)
+    
+    scale_a, scale_b = realize_inputs(scale_a, scale_b)
+
+    # input_nodes: tuple[Any, ...]
+    # workaround for Inductor not supporting optional tensor input arguments
+    # if bias is None:
+
+    if not bias:
+        input_nodes = (mat_a, mat_b, scale_a, scale_b)
+        suffix_args = 2
+    else:
+        bias_arg = realize_inputs(bias)
+        input_nodes = (mat_a, mat_b, scale_a, scale_b, bias_arg)
+        suffix_args = 3
+
+    aten_choice = aten__fp8_mm.bind(
+        input_nodes, layout, out_dtype=out_dtype, use_fast_accum=use_fast_accum
+    )
+
+    choices= []
+    if use_aten_gemm_kernels():
+        choices.append(aten_choice)
+
+    _, is_nonzero = _is_static_problem(layout)
+
+    if is_nonzero and use_triton_template(layout, enable_float8=True):
+        if bias and len(mat_b.get_size()) == len(bias.get_size()) + 1:
+            # Need to unsqueeze bias from [N] -> [1, N]
+            triton_input_nodes = (mat_a, mat_b, scale_a, scale_b, L[aten.unsqueeze](bias, 0))
+        else:
+            triton_input_nodes = input_nodes
+
+        if use_triton_tma_template(mat_a, mat_b) and bias is None:
+            for config in scaled_persistent_mm_configs(m, n, k):
+                kwargs = scaled_mm_options(
+                    config, m, n, k, layout, scale_a, scale_b, use_fast_accum, device_tma=True
+                )
+                scaled_mm_device_tma_template.maybe_append_choice(
+                    choices,
+                    input_nodes=triton_input_nodes,
+                    layout=layout,
+                    workspace_arg=get_tma_workspace_arg(
+                        num_tma_descriptors=2,
+                        device=mat_a.get_device(),
+                    ),
+                    **kwargs,
+                )
+        else:
+            for config in scaled_mm_configs(m, n, k):
+                if k == 16 and config.kwargs["BLOCK_M"] >= 64:
```

Comment:
I dont think these were intended to be mutually exclusive. the tma is intended to be autotune'd with the other configs no ?

Reply:
So I intially added these as mutually exclusive but yeah we likely want to merge before flipping the switch